src/main/java/com/bitmovin/api/http/JsonRestClient.java

```diff
@@ -132,77 +135,141 @@ private void checkOrThrowError(HttpResponse response) throws BitmovinApiExceptio
+    public void delete(URI resource, Map<String, String> headers) throws UnirestException, BitmovinApiException
     {
         HttpRequest request = Unirest.delete(resource.toString())
-                                     .headers(headers)
-                                     .getHttpRequest();
+                .headers(headers)
+                .getHttpRequest();
         this.logHttpRequest(request);
-        HttpResponse<ResponseEnvelope> response =  request.asObject(ResponseEnvelope.class);
+        HttpResponse<ResponseEnvelope> response = executeHttpRequest(request, ResponseEnvelope.class);
         this.logHttpResponse(response);
         this.checkOrThrowError(response);
     }
 
-    public void delete(URI resource, Map<String, String> headers, Object content) throws IOException, RestException, UnirestException, BitmovinApiException
+    public void delete(URI resource, Map<String, String> headers, Object content) throws UnirestException, BitmovinApiException
     {
         if (content == null)
         {
             this.delete(resource, headers);
             return;
         }
         HttpRequest request = Unirest.delete(resource.toString())
-                                     .headers(headers)
-                                     .body(content)
-                                     .getHttpRequest();
+                .headers(headers)
+                .body(content)
+                .getHttpRequest();
         this.logHttpRequest(request);
-        HttpResponse<ResponseEnvelope> response =  request.asObject(ResponseEnvelope.class);
+        HttpResponse<ResponseEnvelope> response = executeHttpRequest(request, ResponseEnvelope.class);
         this.logHttpResponse(response);
         this.checkOrThrowError(response);
     }
 
+    public void setMaxRetries(int maxRetries)
+    {
+        this.maxRetries = maxRetries;
+    }
+
+    public void setMsToWaitForRetrying(int msToWaitForRetrying)
+    {
+        this.msToWaitForRetrying = msToWaitForRetrying;
+    }
+
+    public void setRetryOnError(boolean retryOnError)
+    {
+        this.retryOnError = retryOnError;
+    }
+
+    public void setDebug(boolean debug)
+    {
+        this.debug = debug;
+    }
+
+    private <T> HttpResponse<T> executeHttpRequest(HttpRequest httpRequest, Class<T> responseClass) throws UnirestException
+    {
+        if (this.retryOnError && this.maxRetries > 0 && this.msToWaitForRetrying > 0)
+        {
+            return this.executeHttpRequestWithRetries(httpRequest, responseClass);
+        }
+        return httpRequest.asObject(responseClass);
+    }
+
+    private <T> HttpResponse<T> executeHttpRequestWithRetries(HttpRequest httpRequest, Class<T> responseClass) throws UnirestException
+    {
+        for (int i = 1; i <= this.maxRetries; i++)
+        {
+            HttpResponse<String> httpResponse = httpRequest.asString();
+            if (httpResponse.getStatus() >= 500 && httpResponse.getStatus() < 600)
```

Comment:
LGTM

we get the following error codes:

500:
from the service itself, if it throws an exception

502:
from the google loadbalancer if we have tcp connection issues to envoy (shouldnt happen)

504:
from the google loadbalancer if the envoy, kong, or the service is not available or times out
on GET requests google automatically retries the request and returns with 504 after 60secs

in case of our SQL issue c3p0 does 90 retries each after 1000 millis, thats longer than the http timeout of our loadbalancer, so we are seeing 504 if sql times out